code/game/objects/structures/watercloset.dm

```diff
@@ -357,6 +357,22 @@
 	icon = 'icons/obj/watercloset.dmi'
 	icon_state = "rubberducky"
 	item_state = "rubberducky"
+	attack_verb = list("QUACKED")
+
+/obj/item/bikehorn/New()
+	..()
+	var/datum/component/item_upgrade/I = AddComponent(/datum/component/item_upgrade)
+	I.tool_upgrades = list(
+		UPGRADE_ITEMFLAGPLUS = DUCKING
+	)
+	I.weapon_upgrades = list(
+		GUN_UPGRADE_DUCK = TRUE
+	)
+	I.gun_loc_tag = GUN_MECHANISM
+
+/obj/item/bikehorn/attack_self(mob/user)
+		playsound(src.loc, 'sound/items/duck.ogg', 50, 1)
+		src.add_fingerprint(user)
```

Comment:
```suggestion
/obj/item/bikehorn/rubberducky/New()
	..()
	var/datum/component/item_upgrade/I = GetComponent(/datum/component/item_upgrade)
	I.tool_upgrades[UPGRADE_ITEMFLAGPLUS] = DUCKING
	I.weapon_upgrades[GUN_UPGRADE_DUCK] = TRUE

/obj/item/bikehorn/rubberducky/attack_self(mob/user)
	playsound(loc, 'sound/items/duck.ogg', 50, 1)
	add_fingerprint(user)
```
Overriding the parent item procs makes no sense here, should use those of `/rubberduck`.
Don't create a new item_upgrade component, parent object already got one of these, just fetch and modify it.
Using `src.` should be avoided unless absolutely necessary (it never is).